### PR TITLE
Add more fixtures for agent unit tests.

### PR DESCRIPTION
**Pull request by zulcss**
Add more pytest fixtures to assist in the creation of agent unit
tests the python way.

Signed-off-by: Charles Short <chucks@redhat.com>

### CODE REVIEW

**Review by dbutenhof — COMMENTED** [review]
Seems seriously odd that we have pytest code under `lib/pbench/test/functional/` as well as the neighboring `lib/pbench/test/unit`. They're all unit tests, and the only "functional" unit test appears to be `test_pbench_config.py`

Huh.

File: lib/pbench/test/functional/conftest.py [hunk 1]
```diff
@@ -13,13 +15,48 @@ def setup(request, pytestconfig):
...
+@pytest.fixture
+def pbench_install_dir(tmpdir):
+    """Create an agent install directory for a unittest"""
+    pbench_install_dir = tmpdir / "opt/pbench-agent"
+    os.makedirs(pbench_install_dir)
+    yield pbench_install_dir
+
+
+@pytest.fixture
+def pbench_agent_config(tmpdir, pbench_run, pbench_install_dir, pytestconfig):
```
Comment (portante) [inline lib/pbench/test/functional/conftest.py:43]: We might want to consider reusing what we have, or at least including this in the refactoring: https://github.com/distributed-system-analysis/pbench/blob/master/lib/pbench/test/unit/agent/conftest.py#L8

**Review by portante — COMMENTED** [review]
Can you comment on how we might be able to re-use what we have?

**Comment by portante** [comment]
> Seems seriously odd that we have pytest code under `lib/pbench/test/functional/` as well as the neighboring `lib/pbench/test/unit`. They're all unit tests, and the only "functional" unit test appears to be `test_pbench_config.py`

Could this be resolved by writing a definition of what we mean by "functional" verses "unit" and then reviewing to see which tests fall into which categories?

**Pull request closed without merging**